### PR TITLE
Adds supporting guard form changes for tabs

### LINKED ISSUE #1767: Bug: changes lost when changing tabs from parameters to dependencies
**Describe the bug**
The changes are lost when changing tabs from parameters to dependencies. The "Confirm Navigation" modal is displayed after the navigation. 

**To Reproduce**
1. As the usual user
2. Go to the a parameter page https://dev.algorea.org/en/a/39530140456452546;p=4702,4102,1980584647557587953;a=0/parameters
3. Click add "!" to the title
4. Change to the "dependencies" tab
5. Select "no" into the modal to "Do you want to leave this page and lose its changes?"
6. Check in the parameter tab that the change has been lost

It is important to add a E2E test here to prevent re-breaking that without noticing.

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1767

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

The problem is the tab service switching view directly without router

## Test cases

### CODE REVIEW

**Comment by smadbe** [comment]
[auto-generated message] E2E test report on: https://algorea-static.s3.eu-central-1.amazonaws.com/branch/bugfix/tab-navigation-guard-for-changes/playwright-report/index.html

**Comment by smadbe** [comment]
[auto-generated message] E2E test report on: https://algorea-static.s3.eu-central-1.amazonaws.com/branch/bugfix/tab-navigation-guard-for-changes/playwright-report/index.html

File: src/app/services/tab.service.ts [hunk 1]
```diff
@@ -58,7 +59,20 @@ export class TabService implements OnDestroy {
...
+          destroyed$.next();
+          destroyed$.complete();
+        }
+      }),
+      filter(event => event instanceof NavigationEnd),
+      take(1),
+      takeUntil(destroyed$),
+    ).subscribe(() =>
+      this.activeTab.next(tag)
+    );
```
Comment (smadbe) [inline src/app/services/tab.service.ts:75]: Will `destroyed$` be fully released if there is no navigation cancel event?  I am not sure.

Comment (Iloveall) [reply src/app/services/tab.service.ts]: Fixed

File: src/app/services/tab.service.ts [hunk 2]
```diff
@@ -58,7 +59,21 @@ export class TabService implements OnDestroy {
...
+          destroyed$.complete();
+        }
+      }),
+      filter(event => event instanceof NavigationEnd),
+      takeUntil(destroyed$),
+    ).subscribe(() => {
+      this.activeTab.next(tag);
+      destroyed$.next();
+      destroyed$.complete();
+    });
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/services/tab.service.ts:76]: Yeah, looks much better

**Comment by smadbe** [comment]
[auto-generated message] E2E test report on: https://algorea-static.s3.eu-central-1.amazonaws.com/branch/bugfix/tab-navigation-guard-for-changes/playwright-report/index.html

**Pull request merged**